daemon/daemon_unix.go

```diff
@@ -133,118 +133,137 @@ func (daemon *Daemon) adaptContainerSettings(hostConfig *runconfig.HostConfig, a
 	}
 }
 
-// verifyPlatformContainerSettings performs platform-specific validation of the
-// hostconfig and config structures.
-func verifyPlatformContainerSettings(daemon *Daemon, hostConfig *runconfig.HostConfig, config *runconfig.Config) ([]string, error) {
+func verifyContainerResources(resources *runconfig.Resources) ([]string, error) {
```

Comment:
not sure this should only accept resources :( because I moved the check for `ShmSize` here in https://github.com/docker/docker/pull/18261 and I don't think `ShmSize` is a resource you'll modify with docker set :/ (unless remounting shm but at runtime I don't know how it's feasible)